### PR TITLE
Return database messages such as hints and notices in query results

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

File: lib/postgrex/result.ex [hunk 1]
```diff
@@ -2,21 +2,23 @@ defmodule Postgrex.Result do
...
-    columns:  [String.t] | nil,
-    rows:     [[term] | binary] | nil,
-    num_rows: integer,
-    connection_id: pos_integer}
+          command: atom,
+          columns: [String.t()] | nil,
+          rows: [[term] | binary] | nil,
+          num_rows: integer,
+          connection_id: pos_integer
+        }
```
Comment (ericmj) [inline lib/postgrex/result.ex]: `:messages` type is missing.

File: lib/postgrex/result.ex [hunk 2]
```diff
@@ -2,21 +2,24 @@ defmodule Postgrex.Result do
...
-    columns:  [String.t] | nil,
-    rows:     [[term] | binary] | nil,
-    num_rows: integer,
-    connection_id: pos_integer}
+          command: atom,
+          columns: [String.t()] | nil,
+          rows: [[term] | binary] | nil,
+          num_rows: integer,
+          connection_id: pos_integer,
+          messages: [map()]
```
Comment (michalmuskala) [inline lib/postgrex/result.ex]: If this collects what the database calls "notice", maybe we should also call it "notices"?

**Comment by josevalim** [comment]
I thought about that but notices is not the only kind of message (severity)
that may be returned. So using notice would be too specific. Although I
don’t like messages because it is too generic.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**